src/main/kotlin/lotto/model/InputValidation.kt

```
@@ -0,0 +1,26 @@
+package lotto.model
+
+import lotto.view.Error
+
+object InputValidation {
+    fun validatePurchaseAmount(input: String): Int {
+        val purchaseAmount = input.toIntOrNull() ?: throw IllegalArgumentException(Error.INVALID_INPUT.message)
+        require(purchaseAmount >= 1000) { Error.LOW_PURCHASE_AMOUNT.message }
+        require(purchaseAmount % 1000 == 0) { Error.NOT_DIVISIBLE_BY_1000.message }
+        return purchaseAmount
+    }
+
+    fun validateWinningNumbersInput(numbers: List<Int>) {
+        require(numbers.toSet().size == numbers.size) { Error.DUPLICATE_WINNING.message } // Numbers must be unique
+        require(numbers.all { it in 1..50 }) { Error.OUT_OF_RANGE.message } // any number must be in range 1 - 50
+        require(numbers.size == 6) { Error.INVALID_LOTTO_SIZE.message } // Winning lotto has to be 6 numbers
+    }
+
+    fun validateBonusNumberInput(
+        winningNumbers: List<Int>,
+        bonusNumberInput: Int,
+    ) {
+        require(bonusNumberInput in 1..50) { Error.OUT_OF_RANGE.message } // bonus number must be in range 1 - 50
+        require(!winningNumbers.contains(bonusNumberInput)) { Error.BONUS.message } // bonus number must not be in the winningLotto numbers
```

I think the current code is a great approach!
If the policies for `winningNumber`, `bonusNumber`, and `purchase` change independently in different scenarios, the current code could become a maintenance burden.
But if the entire Lotto game rules tend to change frequently all together, then the current code would actually be excellent for maintenance.

As I mentioned before, whether the current code is beneficial or detrimental for maintenance depends on the business characteristics.

On a separate note, since you've experimented with this approach, I'd recommend trying missions where policies are separated by domain models as well.
I believe you need to experience both sides to understand their respective advantages and disadvantages, which will help you make better decisions quickly when new requirements come up.

---

Initially, I thought having validation in two layers was a better practice, one in InputValidation to catch errors early in the input phase, and another in the domain classes (like Lotto) to guarantee object integrity regardless of where or how they're created.

My idea was to fail fast on user input, but also make sure invalid domain objects can never exist, even if created programmatically elsewhere (e.g. in tests, refactors, or future business logic).

But now I understand your point: if policies change frequently together, then centralizing validation might actually be more maintainable. I really appreciate the suggestion of trying both approaches, that makes total sense, and I’ll definitely experiment with separating validation per model in the next mission.

Thanks again, I'm learning a lot from this process! 🙏